### PR TITLE
Create models

### PR DESCRIPTION
**Pull request by Leeoasis**
### Completed tasks in this PR:

- [ ] Generate models
- [ ] Set up associations between models.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  belongs_to :post
+
+  scope :update_comment_counter, -> { Post.find(3).update(comment_counter: Comment.where(post_id: 3).size) }
```
Comment (elisha2kyakpo1) [inline app/models/comment.rb]: - [ ] Great job using a scope here and in other models, but should understand that we are creating counters for all comments written by users on different posts, and not just a specific post. There are so many Ruby methods you can use to increment the counters. The [icrement](https://apidock.com/rails/ActiveRecord/Base/increment!) function is the best choice for the job.
__HINT:__
```ruby
def update_comments_counter
  post.increment!(:comments_counter)
end
```

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  belongs_to :post
+
+  scope :update_post_likes, -> { Post.find(2).update(likes_counter: Like.where(post_id: 2).size) }
```
Comment (elisha2kyakpo1) [inline app/models/like.rb]: - [ ] Great job using a scope here and in other models, but you should understand that we are creating counters for all likes of users on different posts, and not just a specific post. There are so many Ruby methods you can use to increment the counters. The [icrement](https://apidock.com/rails/ActiveRecord/Base/increment!) function is the best choice for the job.
__HINT:__
```ruby
def update_likes_counter
  post.increment!(:likes_counter)
end
```

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments
+  has_many :likes
+
+  scope :update_posts_counter, -> { User.find(3).update(posts_counter: User.find(3).posts.count) }
```
Comment (elisha2kyakpo1) [inline app/models/post.rb]: - [ ] Great job using a scope here and in other models, but you should understand that we are creating counters for all posts written by different users, and not one user. There are so many Ruby methods you can use to increment the counters. The [icrement](https://apidock.com/rails/ActiveRecord/Base/increment!) function is the best choice for the job.
__HINT:__
```ruby
def update_posts_counter
  post.increment!(:posts_counter)
end
```

File: app/models/post.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  scope :recent_comments, -> { Comment.where(post_id: 2).order(created_at: :desc).first(5) }
```
Comment (elisha2kyakpo1) [inline app/models/post.rb]: - [ ] Kindly do not tie the recent comments to a second post only. we can get our recent comments by removing the where statement.

__HINT:__
```ruby
scope :recent_comments, -> { Comment.order(created_at: :desc).first(5) }
```

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  has_many :likes, foreign_key: :author_id, dependent: :destroy
+  has_many :comments, foreign_key: :author_id, dependent: :destroy
+  has_many :posts, foreign_key: :author_id, dependent: :destroy
+
+  scope :user_recent_posts, -> { find(2).posts.order(created_at: :desc).first(3) }
```
Comment (elisha2kyakpo1) [inline app/models/user.rb]: - [ ] Kindly do not tie the recent posts to a second user only. we can get our recent posts by removing the find method.

__HINT:__
```ruby
scope :user_recent_posts, -> { posts.order(created_at: :desc).first(3) }
```

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @Leeoasis, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
👏 Model associations are nicely set up.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 2 above)
Comment (Leeoasis) [reply app/models/like.rb]: I just made these changes😁

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @Leeoasis, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**